src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1340,6 +1341,43 @@ public bool WaitForExit(int milliseconds)
             return exited;
         }
 
+        /// <summary>
+        /// Instructs the Process component to wait for the associated process to exit, or
+        /// for the <paramref name="cancellationToken"/> to be canceled.
+        /// </summary>
+        public async Task WaitForExitAsync(CancellationToken cancellationToken = default)
+        {
+            if (!Associated) { throw new InvalidOperationException(SR.NoAssociatedProcess); }
+            if (!EnableRaisingEvents) { throw new InvalidOperationException(SR.EnableRaisingEventsRequired); }
```

Comment:
We probably shouldn't set it for them, but if it would "just work" instead of requiring them set it that'd be great.

Reply:
I agree this behavior of throwing is strange.  While not ideal, I think it'd better to make this set EnableRaisingEvents to true if it's false, and just document clearly that it does so (otherwise, we'll just be forcing the developer to do so themselves).  What's the downside?  Just inconsistency with the Exited property?  If we can't stomach that, then I also agree we could make it work without changing the value actually visible from EnableRaisingEvents.

Reply:
Initially I was worried about changing the value of EnableRaisingEvents automatically on the developer, which is why I went the exception route, but having things just work is appealing.

@scalablecory, are you opposed to Stephen's suggestion of setting EnableRaisingEvents to true?

Reply:
I'm fine setting it to true. I just realized `EnableRaisingEvents` only seems to effect the `Exited` event, not either of the others.

Reply:
I've updated the code to not require the user to set `EnableRaisingEvents`. Doing so introduces a chance for an exception if the process exits before setting `EnableRaisingEvents`, so I've added a comment to the top of the method explaining the set of possible cases and how they're handled.

Feedback and any thoughts are appreciated!